src/app/shared/services/app-config/app-config.service.ts

```diff
@@ -100,12 +100,25 @@ export class AppConfigService extends SyncServiceBase {
       return;
     }
 
+    // some app config properties should be functions, but may be defined as boolean values,
+    // e.g. in template parameter_list to disable a certain property on that template
```

Comment:
nit(blocking): slightly tricky issue here, I can see how the introduction of different config overloads (function or boolean) could create a bit of an issue. 

I think we either need to solve this in one of three ways

1) Accept we want this dual functionality, update the type definitions to specify `boolean | function` and add a method to `castBooleanOrFunction` to handle. This code should be called from where the specific properties are used (e.g. the back-button component) to keep highly specific properties out of the general config service

2) Unify around a single type, insisting either authors define functions or defaults use booleans

3) Use separate properties, one that accepts a function and one that accepts boolean, and add logic where these are accessed to identify which one to use and handle accordingly (can likely assume evaluating both statements assuming sensible defaults are used)